x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/index.tsx

```diff
@@ -477,77 +477,90 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
+  const suppressionGroupByDisabledText = useMemo(() => {
+    if (areSuppressionFieldsDisabledBySequence) {
+      return i18n.EQL_SEQUENCE_SUPPRESSION_DISABLE_TOOLTIP;
+    } else if (isMlRule(ruleType) && noMlJobsStarted) {
+      return i18n.MACHINE_LEARNING_SUPPRESSION_DISABLED_LABEL;
+    } else {
+      return alertSuppressionUpsellingMessage;
+    }
+  }, [
+    alertSuppressionUpsellingMessage,
+    areSuppressionFieldsDisabledBySequence,
+    noMlJobsStarted,
+    ruleType,
+  ]);
+
+  const suppressionGroupByFields = useMemo(() => {
+    if (isEsqlRule(ruleType)) {
+      return esqlSuppressionFields;
+    } else if (isMlRule(ruleType)) {
+      return mlSuppressionFields;
+    } else {
+      return termsAggregationFields;
+    }
+  }, [esqlSuppressionFields, mlSuppressionFields, ruleType, termsAggregationFields]);
 
   /**
    * Component that allows selection of suppression intervals disabled:
    *  - if suppression license is not valid(i.e. less than platinum)
    *  - or for not threshold rule - when groupBy fields not selected
    * - Eql sequence is used
+   * - ML Field information is not available
    */
   const isGroupByChildrenDisabled =
-    areSuppressionFieldsDisabledBySequence || !isAlertSuppressionLicenseValid || isThresholdRule
-      ? false
-      : !groupByFields?.length;
+    areSuppressionFieldsDisabled || (isThresholdRule ? false : !groupByFields?.length);
 
   /**
    * Per rule execution radio option is disabled
    *  - if suppression license is not valid(i.e. less than platinum)
    *  - always disabled for threshold rule
    *  - Eql sequence is used and suppression fields are in the default state
+   * - ML Field information is not available
    */
-  const isPerRuleExecutionDisabled =
-    areSuppressionFieldsDisabledBySequence || !isAlertSuppressionLicenseValid || isThresholdRule;
+  const isPerRuleExecutionDisabled = areSuppressionFieldsDisabled || isThresholdRule;
 
   /**
    * Per time period execution radio option is disabled
    *  - if suppression license is not valid(i.e. less than platinum)
    *  - disabled for threshold rule when enabled suppression is not checked
    * - Eql sequence is used and suppression fields are in the default state
+   * - ML Field information is not available
    */
   const isPerTimePeriodDisabled =
-    areSuppressionFieldsDisabledBySequence ||
-    !isAlertSuppressionLicenseValid ||
-    (isThresholdRule && !enableThresholdSuppression);
+    areSuppressionFieldsDisabled || (isThresholdRule && !enableThresholdSuppression);
 
   /**
    * Suppression duration is disabled when
    *  - if suppression license is not valid(i.e. less than platinum)
    *  - when suppression by rule execution is selected in radio button
    *  - when threshold suppression is not enabled and no group by fields selected
    * -  Eql sequence is used and suppression fields are in the default state
+   * - ML Field information is not available
    * */
   const isDurationDisabled =
```

Comment:
Duration for threshold rule is not disabled, when checkbox to enabled suppression is not checked

<img width="940" alt="Screenshot 2024-07-17 at 12 06 55" src="https://github.com/user-attachments/assets/5675fb1a-eff1-4978-8c4a-75d57563c3c3">

Reply:
In the default state (just selecting a threshold rule) this doesn't occur, but I was able to reproduce by:

1. Selecting a "group by" field in the Query Rule tab
2. Selecting the Threshold rule type

@vitaliidm I believe this is related to a previous conversation we had: on [this line](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/index.tsx#L539), `groupByFields.length` won't be nonzero in the above case, because `groupByFields` is not used (nor reset) by threshold rules.

This PR doesn't introduce the behavior we're describing, but we can certainly address it now.

Reply:
@rylnd, thanks, would be great if you can fix it in this PR.
Otherwise, I can create a separate issue and address it later

Reply:
@vitaliidm this was addressed in d2282c914a55352f31dfbb44af3ea33d718c703e

Reply:
thanks for fixing this one